src/v/datalake/record_multiplexer.cc

```diff
@@ -50,6 +50,10 @@ record_multiplexer::operator()(model::record_batch batch) {
           first_timestamp + record.timestamp_delta()};
         kafka::offset offset{batch.base_offset()() + record.offset_delta()};
         int64_t estimated_size = key.size_bytes() + val.size_bytes();
+        chunked_vector<std::pair<iobuf, iobuf>> header_kvs;
+        for (auto& hdr : record.headers()) {
+            header_kvs.emplace_back(hdr.share_key(), hdr.share_value());
```

Comment:
Do we need to distinguish between the `null` key/value vs empty key/value? IIRC the key_length and value_length are < 0 when null, == 0 for empty.

Reply:
Ahh thanks! Will make a `share_key_opt()` and `share_val_opt()` in model::record so it's less inconvenient / easier to remember